vendor-extra/ContentPageBundle/src/Controller/ContentController.php

```diff
@@ -36,37 +39,88 @@ public function __construct(
-                ]
-            )
-            ->then(
-                function (ResponseInterface $response) use ($request) : Response {
-                    $document = FluentDOM::load((string) $response->getBody());
-
-                    $context = [
-                        'lang' => $request->getLocale(),
-                        'dir' => text_direction($request->getLocale()),
-                    ];
-
-                    $event = new CreateContentPageEvent($document, $context);
-                    $this->dispatcher->dispatch($event::NAME, $event);
-
-                    return new Response(
-                        $this->twig->render(
-                            $this->template,
-                            $event->getContext() + ['title' => $event->getTitle(), 'content' => $event->getContent()]
-                        )
-                    );
-                }
-            )
-            ->wait();
+        if (!$id) {
+            return $this->client
+                ->requestAsync(
+                    'GET',
+                    "{$this->service}/items",
+                    [
+                        'headers' => ['Accept' => 'application/xml'],
+                        'http_errors' => true,
+                    ]
+                )
+                ->then(
+                    function (ResponseInterface $response) use ($request) : Response {
+                        $document = FluentDOM::load((string) $response->getBody());
+                        $document->registerNamespace('libero', 'http://libero.pub');
+
+                        $context = [
+                            'lang' => $request->getLocale(),
+                            'dir' => text_direction($request->getLocale()),
+                        ];
+
+                        $list = array_map(function (Element $element) use ($request, $context) {
+                            return $this->client
+                                ->requestAsync(
+                                'GET',
+                                "{$element->getAttribute('service')}/items/{$element->getAttribute('id')}/versions/latest"
+                                )
+                                ->then(
+                                    function (ResponseInterface $response) use ($request, $context) : Document {
+                                        return FluentDOM::load((string) $response->getBody());
+                                    }
+                                );
+                        }, iterator_to_array($document->xpath()->evaluate('libero:item-ref')));
+
+                        $event = new CreateHomePageEvent($list, $context);
+                        $this->dispatcher->dispatch($event::NAME, $event);
+
+                        return new Response(
+                            $this->twig->render(
+                                $this->template,
+                                $event->getContext() + ['title' => 'Home', 'content' => $event->getContent()]
```

Comment:
Justing setting the title manually for now will use the approach we have taken for some headings and aria-labels.